src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -83,12 +86,17 @@ public function __construct(array $normalizers = [], array $encoders = [])
                 $normalizer->setNormalizer($this);
             }
 
+            if ($normalizer instanceof NormalizerInterface) {
+                $this->normalizers[] = $normalizer;
+            }
+            if ($normalizer instanceof DenormalizerInterface) {
+                $this->denormalizers[] = $normalizer;
+            }
             if (!($normalizer instanceof NormalizerInterface || $normalizer instanceof DenormalizerInterface)) {
                 @trigger_error(sprintf('Passing normalizers ("%s") which do not implement either "%s" or "%s" has been deprecated since Symfony 4.2.', \get_class($normalizer), NormalizerInterface::class, DenormalizerInterface::class), E_USER_DEPRECATED);
                 // throw new \InvalidArgumentException(\sprintf('The class "%s" does not implement "%s" or "%s".', \get_class($normalizer), NormalizerInterface::class, DenormalizerInterface::class));
             }
         }
-        $this->normalizers = $normalizers;
```

Comment:
I think we need to keep all the code infrastructure related to this property to keep BC. Then we'll be able to remove the code in 5.0.